### PR TITLE
Allow connect ledger in pass phrase mode

### PR DESCRIPTION
**Pull request by djhahe**
### Checklist (If you won't pass this checklist please comment why)

-   [ ] I removed all commented or unnecessary code.
-   [ ] Provide the screenshots due to UI changed or bug fixed
-   [ ] My code has covered these test cases (please list down your tested cases):

### CODE REVIEW

File: src/screens/home/AccountListScreen/index.tsx [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+  const allAccount = useMemo(() => {
+    return _uniqBy((walletsWithBalance || []).concat(selectedWallet), 'walletInfo.uid').filter(Boolean);
+  }, [walletsWithBalance, selectedWallet]);
+
+  const onDeleteWallet = async (walletInfoDetails: IAccountInfo) => {
+    if (walletInfoDetails.isLedger) {
+      const newLedgerWallets = ledgerWallets?.filter((item) => item.publicKey !== walletInfoDetails.publicKey);
+      await Config.saveItem(Keys.ledgerWallets, newLedgerWallets);
+      refetchLedgerWallets();
```
Comment (0xkynz) [inline src/screens/home/AccountListScreen/index.tsx]: Should `await refetchLedgerWallets` , it is promise function

**Comment by 0xkynz** [comment]
Hi @djhahe, should resolve code smell scanning too

**Comment by 0xkynz** [comment]
Should redirect to "AccountListScreen" when click `Back` button

image

**Comment by 0xkynz** [comment]
Have space in public key of `Account 1`
image

File: src/utils/hooks/useAccountInfo.ts [hunk 2]
```diff
@@ -177,6 +161,20 @@ export const useLedgerAccounts = (
...
+) => {
+  const query = useQuery({
+    queryKey: [ERequestKeys.getLedgerWallets],
+    queryFn: async () => {
+      // get ledger accounts from storage
+      return ((await Config.getItem(Keys.ledgerWallets)) || []) as IAccountInfo[];
+    },
+    ...options,
+  });
+  return { ...query };
```
Comment (0xkynz) [inline src/utils/hooks/useAccountInfo.ts]: Should return `query` instead of spread operation here

**Comment by djhahe** [comment]
> Should redirect to "AccountListScreen" when click `Back` button
> 
> image

This's intentional, to prevent the account list screen stacked 2 times and need to swipe back 2 times to go back to home screen

**Comment by 0xkynz** [comment]
> > Should redirect to "AccountListScreen" when click `Back` button
> > image
> 
> This's intentional, to prevent the account list screen stacked 2 times and need to swipe back 2 times to go back to home screen

Another idea, can add `calbackPage` to this screen:
- When in `Account List` page, set callBack = `AccountListPage` -> Click to back button in "Connect Ledger" redirect by callbackPage
- Similar to homepage

**Review by 0xkynz — APPROVED** [review]

**Comment by djhahe** [comment]
> > > Should redirect to "AccountListScreen" when click `Back` button
> > > image
> > 
> > 
> > This's intentional, to prevent the account list screen stacked 2 times and need to swipe back 2 times to go back to home screen
> 
> Another idea, can add `calbackPage` to this screen:
> 
> * When in `Account List` page, set callBack = `AccountListPage` -> Click to back button in "Connect Ledger" redirect by callbackPage
> * Similar to homepage

The problem with swipe left to back action, it calls `navigation.goBack()` only, for now I cant find way to override the behaviour

**Pull request merged**